### DISCUSSION TITLE
<NuxtLink> to /public content

### DISCUSSION THREAD

**Discussion by AloisSeckar** [nuxt/nuxt#24831] [Questions]
On [my personal website](https://github.com/AloisSeckar/Alois-Seckar.cz) I decided to change all former `<a>` tags with `<NuxtLink>`.

I also have a section with some PDF links (CV + some certificates). I store those files in `/public/pdf` folder.

I thought this would do:

```
<NuxtLink to="/pdf/CV.pdf">
  ŽIVOTOPIS
</NuxtLink>
```

But:
1. it threw a warning to console
![image](https://github.com/nuxt/nuxt/assets/7399922/91370060-df65-4b63-8a41-7fbc4f28b48b)
2. didn't work at all - although it routed to `localhost:3000/pdf/CV.pdf`, it threw 404 error (after refresh the PDF shows).

I had to:
1. add `no-prefetch` option to fix the warning
2. add `external` option to make it work

I can understand, if this is necessary to make it work, but I wanted to make sure, it is intended. If yes, I would like to make a PR for the docs, because I think this should be pointed out. I didn't find it neither [in the docs](https://nuxt.com/docs/api/components/nuxt-link) nor in the [article from Mastering Nuxt website](https://masteringnuxt.com/blog/better-navigation-with-nuxtlink).

P.S. My website [GitHub repo](https://github.com/AloisSeckar/Alois-Seckar.cz), in case it is not indented and it is more likely caused by something in my code.

**Comment by AloisSeckar** [comment]
bump in case someone missed this post 👀

**Comment by TheAlexLichter** [comment]
As you want to treat it like an "external" link, setting `external` seems fine to me 👍🏻 Not sure how it could be detected to "not prefetch" otherwise 🤔 Maybe via regex or similar, but that would be brittle too.

**Reply by AloisSeckar** [reply]
Well, let's  say I don't **mind** it rather than I **want** it...

For me it makes more sense to be "internal" link. because it links to internal content inside my app. But it didn't work. Either I came across some bug, or it is not possible with `/public` content for some reason?

**Reply by TheAlexLichter** [reply]
internal is more meant for SPA-like behavior and routing, which won't apply to a static file

**Reply by AloisSeckar** [reply]
Oh, ok, that makes sense. Thanks for explaining.

**Comment by arashsheyda** [comment]
@manniL how about adding `public` prop to `NuxtLink` which then adds the `no-prefetch` & `external`?
so it's more straightforward and user-friendly

**Reply by TheAlexLichter** [reply]
Feel free to suggest it 👍🏻
I'm not sure if there are enough cases where that extra prop would be used. Also, in theory `external` should be enough, then `no-prefetch` shouldn't be needed 🤔
